eth_account/typed_transactions/typed_transaction.py

```diff
@@ -108,8 +114,11 @@ def from_bytes(cls, encoded_transaction: HexBytes) -> "TypedTransaction":
             raise ValueError("unexpected input")
 
         transaction: Union[
-            "DynamicFeeTransaction", "AccessListTransaction", "BlobTransaction"
-        ]
+            "DynamicFeeTransaction",
+            "AccessListTransaction",
+            "BlobTransaction",
+            "SetCodeTransaction",
+         ]
 
         if encoded_transaction[0] == AccessListTransaction.transaction_type:
```

Comment:
Unrelated but we should probably extract `encoded_transaction[0]` as `tx_type` here by now. We re-use it enough and it makes the code more readable.

```suggestion
        tx_type = encoded_transaction[0]

        if tx_type == AccessListTransaction.transaction_type:
        ... # use tx_type for all cases here
```